code/src/components/app-dropdown/AppDropdown.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useRef, MouseEvent, FocusEvent, ReactNode } from "react";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+import AppDropdownOptions from "@/components/app-dropdown/components/AppDropdownOptions";
+import { AppDropdownProps } from "@/components/app-dropdown/AppDropdown.types";
+
+import "@/components/app-dropdown/AppDropdown.scss";
+
+const AppDropdown = ({
+  options,
+  onSelect,
+  defaultLabel,
+  className
+}: AppDropdownProps) => {
+  const selectedLabelValue = defaultLabel || options[0].label;
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedLabel, setSelectedLabel] =
+    useState<ReactNode>(selectedLabelValue);
+  const dropdownRef = useRef<HTMLElement>(null);
+
+  const iconConditional = isOpen ? (
+    <KeyboardArrowUpIcon fontSize="medium" />
+  ) : (
+    <KeyboardArrowDownIcon fontSize="medium" />
+  );
+
+  const handleSelect = (value: string, label: ReactNode) => {
+    setSelectedLabel(label);
+    setIsOpen(false);
+    onSelect(value);
+  };
+
+  const handleToggleDropdownClick = () => {
+    setIsOpen((prevOpenState) => !prevOpenState);
+  };
+
+  const handleOptionClick = (
+    event: MouseEvent,
+    value: string,
+    label: ReactNode
+  ) => {
+    event.stopPropagation();
+    handleSelect(value, label);
+  };
+
+  const handleBlur = (event: FocusEvent<HTMLElement>) => {
+    if (!dropdownRef.current?.contains(event.relatedTarget as Node)) {
+      setIsOpen(false);
+    }
+  };
+
+  const handleOpenDropdown = () => {
+    return (
+      isOpen && (
+        <AppDropdownOptions
+          handleOptionClick={handleOptionClick}
+          options={options}
+        />
+      )
+    );
+  };
```

Comment:
Don't use function here and call it later. Just create a variable

Reply:
> Don't use function here and call it later. Just create a variable

Can you explain why usage variable better in this case instead function ?

Reply:
You create a function, and then call it, it does not make much sense. For example, you can create variable like following:

```jsx
const a = 1;
```

Or like following:

```jsx
const a = (() => {
    return 1;
})()
```

Both lines produce same output, but first approach is better in terms of readability and it does not add anything to call stack